agent/tool-scripts/datalog/prometheus-metrics-datalog

```diff
@@ -1,35 +1,165 @@
 #!/usr/bin/env python3
 
-import sys, os, json, math
-import subprocess, logging
-from time import time, sleep
+import sys
+import os
+import logging
 
 PROG = os.path.basename(sys.argv[0])
+try:
+    tool_output_dir = sys.argv[1]
+except IndexError:
+    logging.error("[%s]: missing required tool output directory argument",
+            PROG)
+    sys.exit(1)
+else:
+    if not os.path.isdir(tool_output_dir):
+        logging.error("[%s]: the provided tool output directory, '%s', does"
+                " not appear to be a valid directory", PROG, tool_output_dir)
+        sys.exit(1)
 
 try:
-    host = sys.argv[1]
-    interval = int(sys.argv[2])
-    tool_log = sys.argv[3]
-    port = int(sys.argv[4])
-    cert = sys.argv[5]
-    selfsigned = sys.argv[6]
-    key = sys.argv[7]
+    interval_param = sys.argv[2]
 except IndexError:
-    print("{}: missing required arguments".format(PROG), file=sys.stderr)
+    logging.error("[%s]: missing required interval argument", PROG)
     sys.exit(1)
+else:
+    try:
+        interval = int(interval_param)
+    except Exception:
+        logging.error("[%s]: invalid interval argument", PROG)
+        sys.exit(1)
 
-if (cert == "None") or (key == "None"):
-    command = "prom2json http://{}:{:d}/metrics".format(host, port)
+inv_file = os.path.join(tool_output_dir, "inv_hosts")
+
+try:
+    inv_file_param = sys.argv[3]
+except IndexError:
+    logging.error("[%s]: missing required host inventory file", PROG)
+    sys.exit(1)
 else:
-    command = "prom2json{} --cert={} --key={} https://{}:{:d}/metrics".format(
-            " --accept-invalid-cert" if selfsigned == "true" else "", cert,
-            key, host, port)
+    # First, we copy the inventory file provided to our tool output directory
+    # for safe keeping.
+    import shutil
```

Comment:
Lets group the imports all near the top

Reply:
Is there a reason to do that?  By only importing where you need it, we avoid importing too many modules when code paths don't end up using them.

Reply:
https://www.python.org/dev/peps/pep-0008/#imports

Reply:
Let's resolve this when we have the `flake8` and `black` enforcing in place.